app/governance/components/ProposalTable/ProposalTable.tsx

```diff
@@ -0,0 +1,136 @@
+import styles from './ProposalTable.module.scss';
+import { Proposal } from '@/hooks/gov/interfaces/proposal';
+import { formatDate, formatProposalStatus, formatProposalType } from '@/utils/gov/formatData';
+import Text from '@/components/text';
+import Button from '@/components/button/button';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+
+
+interface TableProps {
+  proposals: Proposal[];
+}
+
+const ProposalTable: React.FC<TableProps> = ({ proposals }) => {
+
+  
+  const router = useRouter();
+  const [currentFilter, setCurrentFilter] = useState('All');
+  const [filteredProposals, setFilteredProposals] = React.useState<Proposal[]>(proposals);
+
+  const pageSize = 10;
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(Math.ceil(filteredProposals.length / pageSize));
+
+  useEffect(() => {
+    setTotalPages(Math.ceil(filteredProposals.length / pageSize));
+  }, [filteredProposals.length, pageSize]);
+  console.log(totalPages);
+  const paginatedProposals = filteredProposals.slice(
+    (currentPage - 1) * pageSize,
+    currentPage * pageSize
+  );
+
+  const handlePrevious = () => {
+    if (currentPage > 1) {
+      setCurrentPage(currentPage - 1);
+    }
+  };
+
+  const handleNext = () => {
+    if (currentPage < totalPages) {
+      setCurrentPage(currentPage + 1);
+    }
+  };
+ // console.log(paginatedProposals);
+  useEffect(() => {
+    handleFilterChange(currentFilter);
+    setCurrentPage(1);
+  }, [proposals,currentFilter]);
+
+  const handleFilterChange = (filter: string) => {
+    
+    setCurrentFilter(filter);
+    if (filter === 'All') {
+      setFilteredProposals(proposals);
+    } else {
+      if(filter==='Passed'){
+        setFilteredProposals(proposals.filter(proposal => proposal.status === 'PROPOSAL_STATUS_PASSED'));
+      }
+      if(filter==='Rejected'){
+        setFilteredProposals(proposals.filter(proposal => proposal.status === 'PROPOSAL_STATUS_REJECTED'));
+      }
+      if(filter==='Active'){
+        setFilteredProposals(proposals.filter(proposal => proposal.status === 'PROPOSAL_STATUS_VOTING_PERIOD'));
+      }
+      
+    }
+    
+    //setTotalPages(Math.ceil(filteredProposals.length / pageSize));
+  };
+  
+  const handleRowClick = (proposalId:any) => {
+    // Assuming you have access to the `useRouter` hook here
+    //const router = useRouter();
+  
+    // Navigate to the appropriate page
+    router.push(`/governance/${proposalId}`);
+  };
+  
+  if (!proposals || proposals.length==0) {
+    return <div>Loading Proposals...</div>;
+  }
+  return (
+    <div className={styles.tableContainer}>
+      <Filter onFilterChange={handleFilterChange} currentFilter={currentFilter} />
+      {(filteredProposals.length==0 || !filteredProposals) ? <div className={styles.table}><div className={styles.noProposalContainer}>No {currentFilter} Proposals Available</div></div> : 
+        <table className={styles.table}>
```

Comment:
we have a table component, if it's missing any props please let me know. but would like to be more dependent on our shared components